### PR TITLE
Capability querier

### PR DESCRIPTION
**Pull request by edeyoung**
There are a ton of vendor changes in this PR. To make it easier to view, vendor directories should hopefully all be collapsed (.gitattributes changes should hopefully help with this).
However, be wary of non-vendor files where the diff may have been large enough that github auto-collapsed it (e.g., capabilityquerier/pkg/capabilityquerier/testdata/metadata.json).

----

this PR breaks out the capability querier functionality. It retrieves the capability statements from a list of endpoints and puts the capability statement along with other extracted data onto a message queue.

changes include:
* querying endpoints for json capability statements
* creating a pool of workers to process querying these endpoints. this can be abstracted to be used by other services long term.
* adds a new set of configuration variables for using the queue. added to readme.
* includes rabbitmq information for setting up rabbitmq automatically through docker (lanternmq/definitions.json)
* mock queue for unit tests
* renamed 'endpoints' to 'networkstatsquerier'
* removed the capability querying functionality from the networkstatsquerier
* updated readme with new capabilityquerier info and the networstatsquerier name change
* updated makefile and .travis.yml file to use same testing commands.

----

Pull requests into this repository require the following checks to be completed by the submitter and reviewers.

**Submitter:**
- [x] Administrative tasks are complete.
  - This pull request describes why these changes were made.
  - The JIRA ticket links to this PR.
  - The JIRA ticket for this PR is: https://oncprojectracking.healthit.gov/support/browse/LANTERN-82
  - vendor/ directories have been updated
    - New dependencies have been added
    - Local dependencies have been updated
- [x] Tests are complete.
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.
- [x] Code review has been performed.
  - Code diff has been done and checked to ensure only expected code is being committed.
  - A linter for the language has been run.

**Primary Reviewer:** @mayerm94 

- [x] Code works.
  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
- [x] Code review has been performed.
  - Code is maintainable, reusable, efficient, and correct.
  - Code accomplishes the tasks purpose.
  - Code follows style guidance appropriate for the language, including passing linter checks.
  - Code is well commented (which does not mean verbosely commented).
  - vendor/ directories have been updated
    - New dependencies have been added
    - Local dependencies have been updated
- [x] Tests are complete.
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.

**Secondary Reviewer:**

- [x] Code works.
  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
- [x] Code review has been performed.
  - Code accomplishes the tasks purpose.
- [x] Tests are complete.
  - Tests have been run locally and pass.

**Merger:**

Prior to merging, ensure that all checks have been completed. Delete the branch after the merge is complete.

### CODE REVIEW

File: capabilityquerier/pkg/config/config.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	}
+	err = viper.BindEnv("capquery_qname")
+	if err != nil {
+		return err
+	}
+	err = viper.BindEnv("capquery_numworkers")
+	if err != nil {
+		return err
+	}
+	err = viper.BindEnv("capquery_qryintvl") // in minutes
```
Comment (mayerm94) [inline capabilityquerier/pkg/config/config.go]: Should add README with info on these new .env variables and add them to env.sample and .travis.yml if needed

Comment (edeyoung) [reply capabilityquerier/pkg/config/config.go]: added readme. the unit tests don't use these variables, so don't need to be on .travis.yml. the service is not yet dockerized, so doesn't yet need to be in env.sample.

File: capabilityquerier/pkg/capabilityquerier/queueworkers.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+)
+
+// Job contains all of the information for a queue worker to execute the job.
+// A job contains a context and a duration. The job handler is provided a new context
+// for the job based off or the job's provided context and the given duration.
+// TODO: if queue workers make sense in multiple areas of the code, this should
+// be abstracted to 'interface{}' or similar so arbitrary jobs can be sent.
+type Job struct {
+	Context      context.Context
+	Duration     time.Duration
```
[the remark this answers is not in the document]
Comment (edeyoung) [reply capabilityquerier/pkg/capabilityquerier/queueworkers.go]: yes, we will not record responses for things that take longer than the job duration to respond.

File: capabilityquerier/pkg/capabilityquerier/capabilityquerier.go [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+	}
+	req = req.WithContext(ctx)
+
+	// make the request using a JSON mime type to get a JSON response.
+	// track the mime type to provide to the queue as data re the request.
+	tryOtherMimeType := false
+	mimeType := fhir3PlusJSONMIMEType
+	resp, err = requestWithMimeType(req, mimeType, client)
+	if err != nil {
+		// it's possible that the error is due to a 406 Not Acceptable error for the wrong meme type, so we
```
Comment (mayerm94) [inline capabilityquerier/pkg/capabilityquerier/capabilityquerier.go]: ~meme~ mime

Also, should we check for a 406 specifically that way we are not reissuing the request with a different mime-type in the event of 404s,500s etc...?

**Comment by bmath10** [comment]
I ran it and it seems to be working, but it's not getting to the "Done retrieving and sending capability statement information" log statement (line 74 in main.go). I'm not getting an error or anything, so I'm not sure if I need to wait a while or there's something else going on?

**Comment by bmath10** [comment]
I ran `make run_prod` and then ran the capability querier and got "unable to connect to message queue." I'm not sure if this is expected behavior but wanted to double check!

**Comment by edeyoung** [comment]
> I ran `make run_prod` and then ran the capability querier and got "unable to connect to message queue." I'm not sure if this is expected behavior but wanted to double check!

expected - the ports are not exposed in `run_prod`.

File: capabilityquerier/pkg/capabilityquerier/capabilityquerier.go (hunk 3 above)
Comment (edeyoung) [reply capabilityquerier/pkg/capabilityquerier/capabilityquerier.go]: addressed

**Pull request merged**